lib/QuerySwarm.js

```diff
@@ -136,7 +136,7 @@ module.exports = function(redis) {
 					return;
 
 				// acquire the distributed throttle
-				redis.set(self.id + ':throttle', null, 'PX', self.options.throttle, 'NX', function(err, throttle) {
+				self.redlock.lock(self.id + ':throttle', self.options.throttle, function(err, throttle) {
```

Comment:
node_redis kept complaining with 

> node_redis: Deprecated: The SET command contains a "null" argument.
> This is converted to a "null" string now and will return an error from v.3.0 on. 
> Please handle this in your code to make sure everything works as you intended it to.

and I figured using `redlock.lock` was functionally equivalent